### PR TITLE
Refactor classname splitting from `' '` to `/\s+/`

### PR DESCRIPTION
**Pull request by sakib412**

### CODE REVIEW

**Review by zurfyx — COMMENTED** [review]
Thanks, any real life example when this is needed?

**Comment by sakib412** [comment]
> Thanks, any real life example when this is needed?

https://discord.com/channels/953974421008293909/1180070470935117884/1180070470935117884

image

**Comment by acywatson** [comment]
> Thanks, any real life example when this is needed?

Yea if you have more than one space between classnames, the original implementation will fail. There's a thread about it on Discord.

**Comment by sakib412** [comment]
> Almost feels like this could be extracted out to a util so we don't risk inconsistent class splitting logic in different places.

I have added a util function and testing

File: packages/lexical/src/LexicalUtils.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ import type {
 import type {RootNode} from './nodes/LexicalRootNode';
 import type {TextFormatType, TextNode} from './nodes/LexicalTextNode';
 
+import {splitClasses} from '@lexical/utils';
```
Comment (acywatson) [inline packages/lexical/src/LexicalUtils.ts:31]: Thanks for making this change. However, we don't want the core to have any dependencies, including on other lexical packages, like @lexical/utils.

This means split classes needs to live in the core if it's used there, unfortunately.

**Review by acywatson — CHANGES_REQUESTED** [review]

**Comment by zurfyx** [comment]
Inactivity cleanup. Feel free to reopen

**Pull request closed without merging**